src/shared/type-coercions.js

```diff
@@ -67,3 +67,25 @@ export function toString(value) {
   }
   return '';
 }
+
+/**
+ * @template T
+ * @typedef {import('preact').Ref<T>} Ref
+ */
+
+/**
+ * Helper for downcasting a ref to a more specific type, where that is safe
+ * to do.
+ *
+ * This is mainly useful to cast a generic `Ref<HTMLElement>` to a more specific
+ * element type (eg. `Ref<HTMLDivElement>`) for use with the `ref` prop of a JSX element.
+ * Since Preact only writes to the `ref` prop, such a cast is safe.
+ *
+ * @template T
+ * @template {T} U
+ * @param {Ref<T>|undefined} ref
+ * @return {Ref<U>|undefined}
+ */
+export function downcastRef(ref) {
+  return /** @type {Ref<U>|undefined} */ (ref);
+}
```

Comment:
It might be wise to combine this with the contents of `src/sidebar/util/typing.js` and put it in a separate module from `type-coercions.js`. This helper is similar to `notNull` and different from the others in this file in that it does nothing at runtime, rather it is just a convenient way to assert/cast a type.